test/e2e/containerruntimeconfig_test.go

```diff
@@ -0,0 +1,87 @@
+package e2e_test
+
+import (
+	"fmt"
+	"testing"
+
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/openshift/machine-config-operator/test/e2e/framework"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+)
+
+func TestContainerRuntimeConfigLogSizeMax(t *testing.T) {
+	runTestWithCtrcfg(t, "log-size-max", &mcfgv1.ContainerRuntimeConfiguration{
+		LogSizeMax: resource.MustParse("-1"),
+	})
+}
+
+func TestContainerRuntimeConfigPidsLimit(t *testing.T) {
+	runTestWithCtrcfg(t, "pids-limit", &mcfgv1.ContainerRuntimeConfiguration{
+		PidsLimit: 2048,
+	})
+}
+
+func runTestWithCtrcfg(t *testing.T, testName string, cfg *mcfgv1.ContainerRuntimeConfiguration) {
+	cs := framework.NewClientSet("")
+	testUUID := uuid.NewUUID()
+	matchValue := fmt.Sprintf("%s-%s", testName, testUUID)
+	ctrcfgName := fmt.Sprintf("ctrcfg-%s", testUUID)
+
+	// cache the old machine config value to compare against
+	oldMC := getMcName(t, cs, "worker")
+
+	// label one node from the pool to specify which worker to update
+	unlabelFunc := labelRandomNodeFromPool(t, cs, "worker", mcpNameToRole(matchValue))
+	defer unlabelFunc()
+
+	// create an MCP to match the node we tagged
+	mcpCleanupFunc := createMCP(t, cs, matchValue)
+
+	// create our ctrcfg and attach it to our created node pool
+	cleanupCtrcfg := createCtrcfgWithConfig(t, cs, ctrcfgName, matchValue, cfg)
```

Comment:
why is this called cleanupCtrcfg?  maybe just generatedCtrcfg?

Reply:
It’s because the function is returning a function which cleanup the resource, pretty standard pattern in golang

Reply:
should be called cleanupCtrcfgFunc to be more clear of its type

Reply:
ok i understand esp with the rename! ty!